core/templates/domain/collection/collection-playthrough.model.spec.ts

```diff
@@ -0,0 +1,139 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { CollectionPlaythrough } from './collection-playthrough.model';
+
+/**
+ * @fileoverview Tests for CollectionPlaythroughModel.
+ */
+
+describe('Collection playthrough model', () => {
+  it('should create a new CollectionPlaythrough instance', () => {
+    let collectionPlaythroughObject = new CollectionPlaythrough(
+      'exp_id0', ['exp_id1']);
+    expect(collectionPlaythroughObject.getNextExplorationId())
+      .toEqual('exp_id0');
+    expect(collectionPlaythroughObject.getCompletedExplorationIds())
+      .toEqual(['exp_id1']);
+  });
+
+  it('should create a new CollectionPlaythrough instance from create', () => {
+    let collectionPlaythroughObject = CollectionPlaythrough.create(
+      'exp_id0', ['exp_id1']);
+    expect(collectionPlaythroughObject.getNextExplorationId())
+      .toEqual('exp_id0');
+    expect(collectionPlaythroughObject.getCompletedExplorationIds())
+      .toEqual(['exp_id1']);
+  });
+
+  it('should get the correct next exploration id', () => {
+    var collectionPlaythroughBackendDict = {
+      next_exploration_id: 'exp_id0',
+      completed_exploration_ids: ['exp_id1']
+    };
+    let collectionPlaythroughObject = CollectionPlaythrough
+      .createFromBackendObject(collectionPlaythroughBackendDict);
+    expect(collectionPlaythroughObject.getNextExplorationId())
+      .toEqual('exp_id0');
+  });
+
+  it('should get 1 for next recommended collection node count', () => {
+    var collectionPlaythroughBackendDict = {
+      next_exploration_id: 'exp_id0',
+      completed_exploration_ids: ['exp_id1']
+    };
+    let collectionPlaythroughObject = CollectionPlaythrough
+      .createFromBackendObject(collectionPlaythroughBackendDict);
+    expect(
+      collectionPlaythroughObject.getNextRecommendedCollectionNodeCount()
+    ).toEqual(1);
+  });
+
+  it('should get the completed exploration ids', () => {
+    var collectionPlaythroughBackendDict = {
+      next_exploration_id: 'exp_id0',
+      completed_exploration_ids: ['exp_id1']
+    };
+    let collectionPlaythroughObject = CollectionPlaythrough
+      .createFromBackendObject(collectionPlaythroughBackendDict);
+    expect(
+      collectionPlaythroughObject._completedExplorationIds
+    ).toEqual(['exp_id1']);
+  });
+
+  it('should get the completed exploration node count', () => {
+    var collectionPlaythroughBackendDict = {
+      next_exploration_id: 'exp_id0',
+      completed_exploration_ids: ['exp_id1']
+    };
+    let collectionPlaythroughObject = CollectionPlaythrough
+      .createFromBackendObject(collectionPlaythroughBackendDict);
+    expect(collectionPlaythroughObject.getCompletedExplorationNodeCount())
+      .toEqual(1);
+  });
+
+  it('should get whether the collection is started when no exploration is' +
```

Comment:
Instead of saying 'whether', just state the expected outcome for the test case—e.g. in this case 'should return collection is not started when no exploration is completed'. Ditto below.

Reply:
Done